vendor/engines/secure_rooms/app/controllers/secure_rooms_api/scans_controller.rb

```diff
@@ -8,18 +8,21 @@ class SecureRoomsApi::ScansController < ApplicationController
   before_action :load_user_and_reader
 
   def scan
-    accounts = @user.accounts_for_product(@card_reader.secure_room)
-    requested_id = params[:account_identifier].to_i
-    selected_account = accounts.find { |account| account.id == requested_id }
+    requested_account_id = params[:account_identifier].to_i
```

Comment:
I wanted to be able to just do a `#find` but `accounts_for_product` returns a plain old list.. I'll at least pull it down though